pages/create.js

```diff
@@ -163,12 +163,14 @@ class CreateTournamentForm extends React.Component {
 
     handleGroupSizeInput(input) {
         const newSize = input.target.value;
-        if (newSize <= this.state.groupAdvance) {
-            this.setState({
-                groupSize: newSize, groupAdvance: newSize - 1
-            });
-        } else {
-            this.setState({groupSize: newSize});
+        if (newSize !== undefined && newSize > 2) {
+            if (newSize <= this.state.groupAdvance) {
```

Comment:
Validating the input like following saves you from one indention level for the following default-case code:
```
if (newSize === undefined || newSize < 1) {
  return;
}
```